dotnet/src/webdriver/Remote/ResponseLoggerInterceptor.cs

```diff
@@ -0,0 +1,58 @@
+// <copyright file="ResponseLoggerInterceptor.cs" company="WebDriver Committers">
+// Licensed to the Software Freedom Conservancy (SFC) under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The SFC licenses this file
+// to you under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Net.Http;
+using System.Threading.Tasks;
+using OpenQA.Selenium.Internal.Logging;
+using System;
+
+namespace OpenQA.Selenium.Remote
+{
+    /// <summary>
+    /// Interceptor for logging HTTP responses.
+    /// </summary>
+    public class ResponseLoggerInterceptor : DelegatingHandler
+    {
+        private static readonly ILogger _logger = Log.GetLogger<ResponseLoggerInterceptor>();
+
+        public ResponseLoggerInterceptor(HttpMessageHandler innerHandler)
+            : base(innerHandler)
+        {
+        }
+
+        protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            if (request.Content != null)
+            {
+                string requestContent = await request.Content.ReadAsStringAsync();
+                _logger.Trace($">> Body: {requestContent}");
+            }
+
+            HttpResponseMessage response = await base.SendAsync(request, cancellationToken);
+
+            if ((int)response.StatusCode < 200 || (int)response.StatusCode > 399)
```

Comment:
I am not sure, is there any method available like `Is Successful` rather than compare integers in the range?

Reply:
Let me try to check if there’s something like that and get back to you

Reply:
`IsSuccessStatusCode` - https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpresponsemessage.issuccessstatuscode?view=net-8.0

Reply:
I was about to check on that but thanks for the heads up!

Reply:
So that considers the request successful if the response code was between 200-299 so different to the logic you have above. But maybe that's what we want?

Reply:
yes, we only want to log the responses that aren't `success`.

Reply:
Let's apply the easiest way, and later we always can improve if needed.